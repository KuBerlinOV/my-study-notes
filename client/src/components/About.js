import React from 'react';

const About = () => {
    return (
        <main className='about-page'>
            <h1 className='hd-lg'>About</h1>
            <p class='text'>
                MyStudyNotes is  is a personal project I created during the time of self-study to put into practice what I was learning at the FreeCodeCamp and various other platforms for learning programming.
                Even though, MyStudyNotes was created for practice it also aims to simplify a personal daily task when self-learning.
                MyStudyNotes, therefore allows users to organize their learning material into libraries
                and create notes to save the knowledge and materials accumulated during their self-study.
                It also allows change the relevance of the material written in the note by alternating its status to 'mastered'.
            </p>
        </main>
    )
}

export default About;