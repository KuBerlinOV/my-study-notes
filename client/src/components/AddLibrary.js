import React from 'react';
import { connect } from 'react-redux';
import { startAddLibrary } from '../actions/libraries';
import LibraryForm from './LibraryForm';

export const AddLibrary = (props) => {

    return (
        <div className='add-lib'>
            <h2 className='hd-md'>Create Library</h2>
            <LibraryForm
                handleSubmit={(library) => {
                    props.startAddLibrary(library);
                    props.history.push('/libraries');
                }}
            />
        </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
    startAddLibrary: (library) => dispatch(startAddLibrary(library))
})

export default connect(undefined, mapDispatchToProps)(AddLibrary);