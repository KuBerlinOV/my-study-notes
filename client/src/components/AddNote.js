import React from 'react';
import { connect } from 'react-redux';
import { startAddNote } from '../actions/notes';
import NoteForm from './NoteForm';


export const AddNote = ({ libraryId, startAddNote, handleModal }) => {
    return (

        <NoteForm
            handleModal={handleModal}
            libraryId={libraryId}
            handleSubmit={(note) => {
                startAddNote(note, libraryId);
            }}
        />
    )
}

const mapDispatchToProps = (dispatch) => ({
    startAddNote: (note, libraryId) => dispatch(startAddNote(note, libraryId))
})

export default connect(undefined, mapDispatchToProps)(AddNote);