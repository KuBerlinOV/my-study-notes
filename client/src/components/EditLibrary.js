import React from 'react';
import { connect } from 'react-redux';
import LibraryForm from './LibraryForm';
import { startUpdateLibrary } from '../actions/libraries';


export const EditLibrary = (props) => {
    return (
        <div>
            <LibraryForm
                //library before update
                library={props.library}
                //this is handle submit props
                handleSubmit={(library) => {
                    //dispatch action to edit the expense
                    props.startUpdateLibrary(props.library.id, library);
                    //redirect to dashBoard
                    props.history.push('/libraries')
                }}
            />
        </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
    startUpdateLibrary: (id, library) => dispatch(startUpdateLibrary(id, library))
})

const mapStateToProps = (state, props) => {
    return {
        library: state.libraries.find((library) => library.id == props.match.params.id)
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(EditLibrary);