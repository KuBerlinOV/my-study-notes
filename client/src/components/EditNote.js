import React from 'react';
import { connect } from 'react-redux';
import NoteForm from './NoteForm';
import { startUpdateNote } from '../actions/notes';


export const EditNote = (props) => {
    return (
        <main className='editnote-page'>
            <NoteForm
                //note before update
                note={props.note}
                //this is handle submit props
                handleSubmit={(note) => {
                    //dispatch action to edit the expense
                    props.startUpdateNote(props.note.id, note);
                    //redirect to dashBoard
                    props.history.push('/notes')
                }}
            />
        </main>
    )
}

const mapDispatchToProps = (dispatch) => ({
    startUpdateNote: (id, note) => dispatch(startUpdateNote(id, note))
})

const mapStateToProps = (state, props) => {
    return {
        note: state.notes.find((note) => note.id == props.match.params.id)
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(EditNote);