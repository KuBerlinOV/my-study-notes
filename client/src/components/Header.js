import React from 'react';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';

export const Header = ({ startLogout }) => (
    <header>
        <NavLink activeClassName='is-active' className='nav-link' to='/home' exact={true}>Home</NavLink>
        <NavLink activeClassName='is-active' className='nav-link' to='/notes'>Notes</NavLink>
        <NavLink activeClassName='is-active' className='nav-link' to='/about'>Organizing your studies</NavLink>
        <button onClick={startLogout} >Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => { dispatch(startLogout()) }
})

export default connect(undefined, mapDispatchToProps)(Header);
