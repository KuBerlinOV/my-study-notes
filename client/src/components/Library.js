import React, { useState } from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import Modal from 'react-modal';
import { history } from '../routers/approuter';
import { startRemoveLibrary } from '../actions/libraries';
import Notes from './Notes';

export const Library = ({ topic, description, createdAt, id, tag, startRemoveLibrary, handleLibraryOpen, index }) => {
    // const history = useHistory(); // use this instead of link to redirect to another page with params
    const [openModal, setOpenModal] = useState(false)
    const handleModal = (e) => {
        e.preventDefault();
        setOpenModal(!openModal)
    }

    return (
        <div onClick={() => {
            history.push(`/libraries/${id}`)
            handleLibraryOpen(index)
        }} >
            <h3> Topic: {topic} </h3>
            <p>Description: {description} </p>
            <p>Tag: {tag} </p>
            <p>Date: {moment(createdAt).format('MMMM Do YYYY, h:mm')} </p>
            <button onClick={handleModal}>Delete</button>
            <button onClick={() => {
                history.push(`/editlibrary/${id}`)
            }}> Edit</button>
            <Modal
                isOpen={openModal}
                ariaHideApp={false}
                onRequestClose={handleModal}
                className="delete-warning"
            >
                <h3>Are you sure?</h3>
                <p>Do you really want to delete this Library? This process cannot be undone!</p>
                <button onClick={() => { startRemoveLibrary(id) }}>Delete</button>
            </Modal>
            <Notes libraryId={id} />
        </div>
    )
}



const mapDispatchToProps = (dispatch) => ({
    startRemoveLibrary: (id) => dispatch(startRemoveLibrary(id)),
})

export default connect(undefined, mapDispatchToProps)(Library);