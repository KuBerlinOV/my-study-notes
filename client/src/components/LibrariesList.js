import React from 'react';
import Library from './Library';
import { connect } from 'react-redux';
import PrivateRoute from '../routers/PrivateRoute';
import selectLibraries from '../selectors/libraries';
import LibraryInfoCard from './LibraryInfoCard';
import { Grid } from '@material-ui/core';

const LibrariesList = (props) => {
    return (
        <Grid>
            {props.libraries.length === 0 ? (
                <p>Create your first library and organize your study</p>
            ) : (
                props.libraries.map(library => {
                    return <LibraryInfoCard
                        id={library.id}
                        topic={library.topic}
                        description={library.description}
                        tag={library.tag}
                        createdAt={library.createdAt}
                    />
                })


            )}
            <PrivateRoute path='/libaries/:id' component={Library} />
        </Grid>
    )
}

const mapStateToProps = (state) => {
    return {
        libraries: selectLibraries(state.libraries, state.filters)
    }
}


export default connect(mapStateToProps)(LibrariesList)