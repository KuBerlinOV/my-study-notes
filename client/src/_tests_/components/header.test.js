// import ReactShallowRenderer from 'react-test-renderer/shallow';

import React from 'react';
import { Header } from '../../components/Header';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';



// test('should render header correctly', () => {
//     const renderer = new ReactShallowRenderer();
//     renderer.render(<Header />);
//     expect(renderer.getRenderOutput()).toMatchSnapshot();

//     console.log(renderer.getRenderOutput());
// })



describe('Header', () => {
    it('should render Header correctly', () => {
        const startLogoutSpy = jest.fn();
        const wrapper = mount(<BrowserRouter><Header startLogout={startLogoutSpy} /> </BrowserRouter>)
        expect(wrapper).toMatchSnapshot();
    })
    it('should call startLogout on button click', () => {
        const startLogoutSpy = jest.fn();
        const wrapper = mount(<BrowserRouter><Header startLogout={startLogoutSpy} /> </BrowserRouter>);
        wrapper.find('button').simulate('click');
        expect(startLogoutSpy).toHaveBeenCalled();
    })
})
