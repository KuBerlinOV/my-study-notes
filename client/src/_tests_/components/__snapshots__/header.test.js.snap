// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header
      startLogout={[MockFunction]}
    >
      <header>
        <NavLink
          activeClassName="is-active"
          className="nav-link"
          exact={true}
          to="/home"
        >
          <Link
            aria-current={null}
            className="nav-link"
            to={
              Object {
                "hash": "",
                "pathname": "/home",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="nav-link"
              href="/home"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/home"
                onClick={[Function]}
              >
                Home
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <NavLink
          activeClassName="is-active"
          className="nav-link"
          to="/notes"
        >
          <Link
            aria-current={null}
            className="nav-link"
            to={
              Object {
                "hash": "",
                "pathname": "/notes",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="nav-link"
              href="/notes"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/notes"
                onClick={[Function]}
              >
                Notes
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <NavLink
          activeClassName="is-active"
          className="nav-link"
          to="/about"
        >
          <Link
            aria-current={null}
            className="nav-link"
            to={
              Object {
                "hash": "",
                "pathname": "/about",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="nav-link"
              href="/about"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/about"
                onClick={[Function]}
              >
                Organizing your studies
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <button
          onClick={[MockFunction]}
        >
          Logout
        </button>
      </header>
    </Header>
     
  </Router>
</BrowserRouter>
`;
